Dockerfile.ci-rp

```diff
@@ -0,0 +1,48 @@
+ARG REGISTRY
+ARG VERSION
+
+###############################################################################
+# builder is responsible for all compilation and validation of the RP
+###############################################################################
+FROM ${REGISTRY}/ubi8/go-toolset:1.20.10 as builder
```

Comment:
go toolset has been bumped recently to 1.20.12-5 in another [PR](https://github.com/Azure/ARO-RP/pull/3532/files) which has been merged, it is probably better to bump it here as well.